src/exception/treatment-detail-exceptions/repo.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import mongoose from 'mongoose';
+import { loggerFor } from '../../logger';
+import { Result, failure, success } from '../../utils/results';
+import { TreatmentDetailException, TreatmentDetailExceptionSummary } from './model';
+
+const L = loggerFor(__filename);
+
+const treatmentDetailExceptionSchema = new mongoose.Schema<TreatmentDetailException>({
+	programId: String,
+	donorSubmitterIds: [String],
+});
+
+// check if model exists already to account for file watchers eg. test runner with live reload
+const TreatmentDetailExceptionModel = mongoose.model<TreatmentDetailException>(
+	'TreatmentDetailException',
+	treatmentDetailExceptionSchema,
+);
+
+/**
+ * Save the provided exception. If an exception for this program exists then it will be overwritten with this value.
+ * Otherwise a new TreatmentDetailException will be created in the database.
+ *
+ * A return with failure state indicates a failure communicating with the database
+ * @param treatmentDetailExceptionException
+ * @return Result that will indicate if the save happened successfully. If so, a copy of the now saved data will be returned.
+ */
+export const createOrUpdate = async (
+	treatmentDetailException: TreatmentDetailException,
+): Promise<Result<TreatmentDetailException>> => {
+	try {
+		const updatedException = await TreatmentDetailExceptionModel.findOneAndReplace(
+			{ programId: treatmentDetailException.programId },
+			treatmentDetailException,
+			{ upsert: true, new: true },
+		).lean(true);
+
+		return success(updatedException);
+	} catch (error) {
+		L.error(
+			`Failure creating or updating Missing Entity Exception for program: ${treatmentDetailException.programId}`,
+			error,
+		);
+		return failure(`Unable to save Missing Entity Exception to database.`);
```

Comment:
Missed replacement from copy paste 😛

Reply:
Haha I tried to be as thorough as I could, can't believe I missed these!
Updated